common/unsafe/src/main/java/org/apache/spark/unsafe/UnsafeAlignedOffset.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.unsafe;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Class to make changes to record length offsets uniform through out
+ * various areas of Apache Spark core and unsafe.  The SPARC platform
+ * requires this because using a 4 byte Int for record lengths causes
+ * the entire record of 8 byte Items to become misaligned by 4 bytes.
+ * Using a 8 byte long for record length keeps things 8 byte aligned.
+ */
+
+public class UnsafeAlignedOffset {
+
+  private static int uaoSize;
+  static final List<String> MISALIGNED_ARCH = Arrays.asList("sparc", "sparcv9");
+
+  static {
+    if (MISALIGNED_ARCH.contains(System.getProperty("os.arch"))) {
+      uaoSize = 8;
+    } else {
+      uaoSize = 4;
+    }
+  }
+
+  public static int getUaoSize() {
+    return uaoSize;
+  }
+
+  public static int getSize(Object object, long offset) {
+    if (uaoSize == 4) {
+      assert(offset % 4 == 0);
```

Comment:
It's not clear this should be an `assert`; it will be disabled by default at runtime. Do you need it on only for testing?